File: 2016/d06_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative 'd06'

class TestD06 < Minitest::Test
  def setup
    @example = <<~EXAMPLE
      eedadn
      drvtee
      eandsr
      raavrd
      atevrs
      tsrnev
      sdttsa
      rasrtv
      nssdts
      ntnada
      svetve
      tesnvt
      vntsnd
      vrdear
      dvrsen
      enarar
    EXAMPLE
  end

  def test_part1
    assert_equal 'easter', D06.new(@example).part1
  end

  def test_part2
    assert_equal 'advent', D06.new(@example).part2
  end
end

File: 2018/test/aoc2018/d06_test.clj
(ns aoc2018.d06-test
  (:require [clojure.test :refer [deftest is]]
            [aoc2018.d06 :as d]))

(def example "1, 1\n1, 6\n8, 3\n3, 4\n5, 5\n8, 9")

(deftest part1-test
  (is (= 17 (d/part1 example))))

(deftest part2-test
  (is (= 16 (d/part2 example 32))))
